Sources/CardPayments/CardClient.swift

```diff
@@ -154,6 +164,43 @@ public class CardClient: NSObject {
         return url.queryItems?.first { $0.name == param }?.value
     }
 
+    private func startVaultThreeDSecureChallenge(
+        url: URL,
+        setupTokenID: String
+    ) {
+
+        vaultDelegate?.cardThreeDSecureWillLaunch(self)
+
+        webAuthenticationSession.start(
+            url: url,
+            context: self,
+            sessionDidDisplay: { [weak self] didDisplay in
+                if didDisplay {
+                    // should we send different analytics for vault 3DS?
+                    self?.analyticsService?.sendEvent("card-payments:3ds:challenge-presentation:succeeded")
+                } else {
+                    self?.analyticsService?.sendEvent("card-payments:3ds:challenge-presentation:failed")
+                }
+            },
+            sessionDidComplete: { url, error in
+                self.vaultDelegate?.cardThreeDSecureDidFinish(self)
+                if let error = error {
+                    switch error {
+                    case ASWebAuthenticationSessionError.canceledLogin:
+                        self.notifyVaultCancellation()
+                        return
+                    default:
+                        self.notifyVaultFailure(with: CardClientError.threeDSecureError(error))
+                        return
+                    }
+                }
+
+                let cardVaultResult = CardVaultResult(setupTokenID: setupTokenID, status: "3dsVerificationAttempted", deepLinkURL: url)
```

Comment:
* This boolean property would be in addition to `status`

Reply:
That's true. I was just talking to Nirvana about that. The only thing is that status value is  unknown in our SDK after 3DS verification. That's why we've considered "unknown" or something else.

Reply:
> So what should we do about status when updated status is not known after 3DS verification?
keep it optional and return boolean true?

If we don't know the status at this point in time since it's not returned in the URL, we shouldn't return anything for status to the merchant. So yeah I think going with your approach of keeping it optional and returning a bool is the best we can do for merchants. Hopefully as time goes on we can get feedback from merchants on improving the experience and relay those to the API team.

Reply:
That's a good call.

Reply:
+1 to Jax's comment - if we don't have a `status` from the API we hit, we shouldn't set it

Reply:
Yes, Steven brought this up as well. I agree. I was wanting to clarify that this state is not updated
after 3ds verification, so I thought of "UNKOWN" or something else. 
But setting it nil makes sense.